### PR TITLE
Add monitors.d directory + config to default configuration

### PR DESCRIPTION
**Pull request by andrewvc**
This makes heartbeat much easier to use by default, and less confusing in terms of where/how
to create this directory.

### CODE REVIEW

**Comment by andrewvc** [comment]
For any reviewers out there, how can we test this in the context of apt/deb to make sure those packages add the same files to the directory layout?

**Comment by ruflin** [comment]
@andrewvc Anything I can help with to push this PR forward?

**Comment by andrewvc** [comment]
@ruflin not quite ready for review yet. I just rebased it and added support in the magefile.
I still need to write tests (and manually test it).

Also fixed heartbeat.yml to point at a relative path

**Comment by andrewvc** [comment]
OK, things seem to work and package tests are added. I actually haven't manually tested them yet. Can do tomorrow.

There's a lot of opportunity to DRY things up in the package tests, but it feels premature to me for the effort. WDYT @ruflin / @andrewkroh ?

**Comment by ruflin** [comment]
@andrewvc Code LGTM. Would be good if you could also test it locally and after merging keep an eye on the packaging tests. 

In case you rebase again on master or do an additonal push, could you also make Hound happy? CI failures are not related.

**Comment by andrewvc** [comment]
@ruflin I tested this on OSX/Ubuntu Server/Fedora VMs using .tar.gz/.deb/.rpm respectively .

Everything worked great. On the linux box I set things up as a service using ES as a service. All I had to do to get a working monitor was remove the `.disabled` from a monitor in the sample dir and customize it to my liking.

I've improved the PR somewhat adding separate sample files per protocol to `monitors.d`.

File: heartbeat/_meta/beat.yml [hunk 1]
```diff
@@ -9,7 +9,17 @@
...
 ############################# Heartbeat ######################################
 
-# Configure monitors
+# Define a directory to load monitor definitions from. Definitions take the form
+# of individual yaml files.
+heartbeat.config.monitors:
+  # Directory + glob pattern to search for configuration files
+  path: ${path.config}/monitors.d/*.yml
+  # If enabled, heartbeat will periodically check the config.monitors path for changes
+  reload.enabled: true
```
Comment (ruflin) [inline heartbeat/_meta/beat.yml]: We have this disabled in other Beats by default: https://github.com/elastic/beats/blob/master/filebeat/_meta/common.p2.yml#L64 If I remember correctly the argument was that we do not want to the user to be surprised if we moves files around that it also has directly and affect on his running Beat by default. I wonder if this also applies here?

Sorry, only spotted this now.

Comment (andrewvc) [reply heartbeat/_meta/beat.yml]: makes sense, I'll change that.

File: CHANGELOG.asciidoc [hunk 2]
```diff
@@ -526,6 +526,8 @@ https://github.com/elastic/beats/compare/v6.4.0...v6.5.0[View commits]
 - Added support for extra TLS/x509 metadata. {pull}7944[7944]
 - Added stats and state metrics for number of monitors and endpoints started. {pull}8621[8621]
 - Add last monitor status to dashboard table. Further break out monitors in dashboard table by monitor.ip. {pull}9022[9022]
+- Add central management support. {pull}9254[9254]
```
Comment (ruflin) [inline CHANGELOG.asciidoc]: Rebase fun

File: heartbeat/_meta/beat.reference.yml [hunk 3]
```diff
@@ -235,11 +234,1150 @@ heartbeat.monitors:
...
-
 heartbeat.scheduler:
   # Limit number of concurrent tasks executed by heartbeat. The task limit if
   # disabled if set to 0. The default is 0.
   #limit: 0
 
   # Set the scheduler it's timezone
   #location: ''
+
+#================================ General ======================================
```
Comment (ruflin) [inline heartbeat/_meta/beat.reference.yml]: I'm trying to understand where this huge diff here is coming from. Was that just missing before?

Comment (andrewvc) [reply heartbeat/_meta/beat.reference.yml]: Good question. I edited `beat.reference.yml` and copied it over to `_meta/beat.reference.yml`, which shouldn't have caused this. Maybe it was out of sync for a while?

Comment (andrewvc) [reply heartbeat/_meta/beat.reference.yml]: AFAICT this is good stuff to have in there at any rate right?

Comment (andrewvc) [reply heartbeat/_meta/beat.reference.yml]: @ruflin I fixed this file, this shouldn't have made it into this PR.

File: CHANGELOG.asciidoc (hunk 2 above)
Comment (andrewvc) [reply CHANGELOG.asciidoc]: Fixed

**Pull request merged**